File: KSNavigationControllerExample/KSNavigationControllerExample/TestViewController.m
//
//  TestViewController.m
//
//  Created by Alex Gordiyenko on 8/1/16.
//  Copyright © 2016 Alex Gordiyenko. All rights reserved.
//

#import "TestViewController.h"

@interface TestViewController ()

@property (weak) IBOutlet NSTextField *textField;

@end

@implementation TestViewController

@synthesize navigationController = _navigationController;

- (IBAction)pushaction:(id)sender {
    TestViewController *testVC = [[TestViewController alloc] init];
    [self.navigationController pushViewController:testVC animated:YES];
}
- (IBAction)popaction:(id)sender {
    [self.navigationController popViewControllerAnimated:YES];
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.textField.stringValue = [@(self.navigationController.viewControllersCount) stringValue];
    
    self.view.wantsLayer = YES;
    self.view.layer.backgroundColor = [NSColor colorWithRed:arc4random_uniform(63) / 63.0 + 0.5 green:arc4random_uniform(63) / 63.0 + 0.5 blue:arc4random_uniform(63) / 63.0 + 0.5 alpha:1].CGColor;
}

- (void)dealloc {
    NSLog(@"%s", __PRETTY_FUNCTION__);
}

@end

File: KSNavigationControllerExampleSwift/KSNavigationControllerExampleSwift/TestViewController.swift
//
//  TestViewController.swift
//  KSNavigationControllerExampleSwift
//
//  Created by Alex on 8/4/16.
//  Copyright © 2016 Alex. All rights reserved.
//

import Cocoa

class TestViewController: NSViewController, KSNavigationControllerCompatible {
    weak var navigationController: KSNavigationController?

    @IBOutlet weak var textField: NSTextField!
    
    @IBAction func pushAction(_ sender: AnyObject) {
        self.navigationController?.pushViewController(TestViewController(), animated: true)
    }

    @IBAction func popAction(_ sender: AnyObject) {
        _ = self.navigationController?.popViewControllerAnimated(true)
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.view.wantsLayer = true
        self.view.layer?.backgroundColor = NSColor(red: CGFloat(arc4random_uniform(63)) / 63.0 + 0.5, green: CGFloat(arc4random_uniform(63)) / 63.0 + 0.5, blue: CGFloat(arc4random_uniform(63)) / 63.0 + 0.5, alpha: 1).cgColor
        
        if let count = self.navigationController?.viewControllersCount {
            self.textField.stringValue = String(count)
        }
    }
}
